### PR TITLE
add negotiate support

### PR DESCRIPTION
**Pull request by nitzmahone**
updated tests to exercise NTLM, Negotiate, and both

### CODE REVIEW

File: requests_ntlm/requests_ntlm.py [hunk 1]
```diff
@@ -86,22 +88,23 @@ def retry_using_http_NTLM_auth(self, auth_header_field, auth_header,
...
 
-        ntlm_header_value = [
-            s for s in (val.lstrip() for val in auth_header_value.split(','))
-            if s.startswith('NTLM ')
-        ][0].strip()
+        auth_strip = auth_type + ' '
+
+        ntlm_header_value = list(filter(
+            lambda s: s.startswith(auth_strip), auth_header_value.split(',')
+        ))[0].strip()
```
Comment (Lukasa) [inline requests_ntlm/requests_ntlm.py]: Rather than create the intervening list here, it might be better to use `next()` to return the first element.

**Comment by Lukasa** [comment]
I've left a note in the diff, but this otherwise looks fine to me. That said, it's a big diff, so I'd like @sigmavirus24 to review as well.

File: requests_ntlm/requests_ntlm.py (hunk 1 above)
Comment (nitzmahone) [inline requests_ntlm/requests_ntlm.py]: Hmm, that's actually a regression from a previous version of requests-ntlm- I initially hacked this onto the raw bits from PyPI and then dumped into a repo later. I'll fix that.

**Comment by nitzmahone** [comment]
OK, updated the auth header selection back to the way it was in master, but switched it to use next() instead of building a list.

**Comment by Lukasa** [comment]
Cool, this looks good to me! Thanks @nitzmahone! :sparkles: :cake: :sparkles:

**Pull request merged**